CI broken

<!-- Welcome! Thank you for contributing. These HTML comments will not render in the issue.

Before creating a new issue:
* Search for relevant issues
* Follow the issue reporting guidelines:
https://jupyterlab.readthedocs.io/en/latest/getting_started/issue.html
-->

## Description

<!--Describe the bug clearly and concisely. Include screenshots if possible-->
Since a day or so, the CI is broken.

The pathologic error seen are:
For JS tests when running `jlpm run build`

```
error TS2688: Cannot find type definition file for 'node'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions
Error: ../observables/src/observablemap.ts(4,29): error TS2307: Cannot find module '@lumino/disposable' or its corresponding type declarations.
```

For other tests

```
$ node scripts/ensure-buildutils.js && node buildutils/lib/ensure-repo.js --force
internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module 'fs-extra'
Require stack:
- /home/runner/work/jupyterlab/jupyterlab/scripts/ensure-buildutils.js
```

```
sh: 1: prettier: not found
```

etc

See for example https://github.com/jupyterlab/jupyterlab/pull/12899/checks

By curiosity I ran a second time the JS tests on the last master commit that passed and it failed (you can see it by switching from attempt 1 and attempt 2): https://github.com/jupyterlab/jupyterlab/actions/runs/3035624109

This calls for an external impact - certainly related to the `pip install -e .[dev,test]` that does not run `jlpm install` on the current folder.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -170,7 +170,7 @@ exclude = [
 ]
 
 [tool.hatch.build.hooks.jupyter-builder]
-dependencies = ["hatch-jupyter-builder>=0.3.2,<0.6.0"]
+dependencies = ["hatch-jupyter-builder>=0.3.2"]
 build-function = "buildapi.builder"
 ensured-targets = [
     "jupyterlab/static/package.json",

